src/main/java/org/jenkinsci/plugins/ssegateway/sse/EventDispatcher.java

```diff
@@ -339,11 +339,33 @@ private void dispatchReload() {
             LOGGER.error("Unable to send reload event to client.", e);
         }
     }
+
+    /**
+     * Called on queue events to validate that the disaptcher is still alive
+     */
+    private void validateDispatcher() {
+        Retry retry = retryQueue.peek();
+        if (retry != null) {
+            long ctime = System.currentTimeMillis();
+            long retry_age = (ctime - retry.timestamp);
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug(String.format("EventDispatcher (%s) - timestamp: %d - current: %d - age: %d", this, retry.timestamp, ctime, retry_age));
+            }
+            // check time of oldest retry envent
+            if (retry_age > RETRY_QUEUE_EVENT_LIFETIME) {
+                // oldest event has timed out - remove all events from retry queue
+                LOGGER.debug("EventDispatcher {} processRetries - clear retryQueue", this);
+                retryQueue.clear();
+            }
+        }
+        checkDispatcherFailTimeout("dispatcher.validation");
+    }
     
     private void addToRetryQueue(@Nonnull Message message) {
         // check retry queue is empty
         //  -> we are adding the first element
         //  -> start the retryqueue timer
+        validateDispatcher();
```

Comment:
Check the queue before adding so retries are re-scheduled

Reply:
nice comments! 
maybe good to have those comments in the javadoc or in the code?